WireDomain/Sources/WireDomain/Notifications/Builders/NewMessageNotificationBuilder.swift

```diff
@@ -0,0 +1,378 @@
+        let conversation = await conversationLocalStore.fetchOrCreateConversation(
+            id: conversationID.uuid,
+            domain: conversationID.domain
+        )
+
+        let sender = await userLocalStore.fetchOrCreateUser(
+            id: senderID.uuid,
+            domain: senderID.domain
+        )
+
+        let senderName = await userLocalStore.name(for: sender)
+        let conversationName = await conversationLocalStore.name(for: conversation)
+        let isGroupConversation = await conversationLocalStore.isGroupConversation(conversation)
+        let selfUser = await userLocalStore.fetchSelfUser()
+        let teamName = await userLocalStore.teamName(for: selfUser)
+        let isMessageSilenced = await conversationLocalStore.isMessageSilenced(
+            message,
+            senderID: senderID.uuid,
+            conversation: conversation
+        )
+        let selfUserID = await userLocalStore.id(for: selfUser)
+        let shouldHideNotification = await conversationLocalStore.shouldHideNotification()
+
+        self.context = Context(
+            senderName: senderName,
+            conversationName: conversationName,
+            isGroupConversation: isGroupConversation,
+            teamName: teamName,
+            isMessageSilenced: isMessageSilenced,
+            conversationID: conversationID,
+            senderID: senderID.uuid,
+            selfUserID: selfUserID,
+            hidesNotificationContent: shouldHideNotification
+        )
+    }
+
+    func shouldBuildNotification() async -> Bool {
+        !context.isMessageSilenced
+    }
+
+    func buildContent() async -> UNMutableNotificationContent {
+        guard !context.hidesNotificationContent else {
+            return buildHiddenNotification()
+        }
+
+        switch message.content {
+        case .location:
+            return buildLocationNotification()
+        case .knock:
+            return buildPingNotification()
+        case .image:
+            return buildAssetNotification(ofType: .image)
+        case let .ephemeral(ephemeral):
+            return await buildEphemeralNotification(ephemeral: ephemeral)
+        case let .text(text):
+            return await buildTextNotification(text)
+        case let .composite(composite):
+            let text = composite.items.compactMap(\.text).first
+            return await buildTextNotification(text)
+        case let .asset(assetData):
+            switch assetData.original.metaData {
+            case .audio:
+                return buildAssetNotification(ofType: .audio)
+            case .video:
+                return buildAssetNotification(ofType: .video)
+            case .image:
+                return buildAssetNotification(ofType: .image)
+            default:
+                return buildAssetNotification(ofType: .fileUpload)
+            }
+        case .hidden:
+            return buildHiddenNotification()
+        default:
+            return UNMutableNotificationContent()
```

Comment:
It would be clearer if we returned `nil`, assuming that we don't want to show any notification in this case.